src/FixMath_Autotests.h

```diff
@@ -37,6 +37,20 @@ namespace FixMathPrivate {
       static_assert((e + e).getNF() == 2, "test fail");
       static_assert((d + e).getNI() == 11, "test fail");
 
+      // Two's complement peculiar additions and opposite
+      static_assert(SFixAuto<-128>().getNI() == 7, "test fail");
+      static_assert(SFixAuto<127>().getNI() == 7, "test fail");
+      static_assert(SFixAuto<128>().getNI() == 8, "test fail");
+
+#if __cplusplus >= 202002L
```

Comment:
That will effectively disable the check. The funny thing is that the Uno's AVR-GCC, although even older, accepts the checks as constexpr.

Thus, I suggest to change the guard to
```#if defined(__GNUC__) || (__cplusplus >= 202002L)```
(of course this applies to the other section, where this is used in the tests, too).

Reply:
Good idea, thanks. I added it because the STM32 compilation would fail (even though UNO would succeed…)

Reply:
For some reason, this does not work…